components/org.wso2.micro.integrator.extensions/org.wso2.micro.integrator.management.apis/src/main/java/org/wso2/micro/integrator/management/apis/UserResource.java

```diff
@@ -0,0 +1,162 @@
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.wso2.micro.integrator.management.apis;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.apache.synapse.MessageContext;
+import org.apache.synapse.config.SynapseConfiguration;
+import org.json.JSONArray;
+import org.json.JSONObject;
+import org.wso2.micro.integrator.security.MicroIntegratorSecurityUtils;
+import org.wso2.micro.integrator.security.user.api.RealmConfiguration;
+import org.wso2.micro.integrator.security.user.api.UserStoreException;
+import org.wso2.micro.integrator.security.user.api.UserStoreManager;
+
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.Objects;
+import java.util.Set;
+
+import static org.wso2.micro.integrator.management.apis.Constants.BAD_REQUEST;
+import static org.wso2.micro.integrator.management.apis.Constants.INTERNAL_SERVER_ERROR;
+import static org.wso2.micro.integrator.management.apis.Constants.IS_ADMIN;
+import static org.wso2.micro.integrator.management.apis.Constants.NOT_FOUND;
+import static org.wso2.micro.integrator.management.apis.Constants.ROLES;
+import static org.wso2.micro.integrator.management.apis.Constants.STATUS;
+import static org.wso2.micro.integrator.management.apis.Constants.USERNAME_PROPERTY;
+import static org.wso2.micro.integrator.management.apis.Constants.USER_ID;
+
+
+/**
+ * Resource for login. This obtains JWT token. This is basic auth protected.
+ */
+public class UserResource implements MiApiResource {
+
+    private static final Log LOG = LogFactory.getLog(UserResource.class);
+
+    // HTTP method types supported by the resource
+    Set<String> methods;
+
+    public UserResource() {
+
+        methods = new HashSet<>();
+        methods.add(Constants.HTTP_GET);
+        methods.add(Constants.HTTP_DELETE);
+    }
+
+    @Override
+    public Set<String> getMethods() {
+        return methods;
+    }
+
+    @Override
+    public boolean invoke(MessageContext messageContext, org.apache.axis2.context.MessageContext axis2MessageContext,
+                          SynapseConfiguration synapseConfiguration) {
+        LOG.info("Handling request by user resource");
+        String httpMethod = axis2MessageContext.getProperty(Constants.HTTP_METHOD_PROPERTY).toString();
+        JSONObject response;
+        try {
+            switch (httpMethod) {
+                case Constants.HTTP_GET: {
+                    response = handleGet(messageContext);
+                    break;
+                }
+                case Constants.HTTP_DELETE: {
+                    response = handleDelete(messageContext);
+                    break;
+                }
+                default: {
+                    response = Utils.createJsonError("Unsupported HTTP method, " + httpMethod + ". Only GET, POST, "
```

Comment:
The request won't reach here if the methods are not supported. It would be stopped in mediation. Hence we may leave this case empty or just use an else if.

Reply:
If by any chance the request is forwarded upto this point, it'll throw this error, I see no harm in having it. Else if will complicate things if a new method gets introduced.

Reply:
Yes no harm in having this . I was thinking that since this can't reach here logically and if we remove this, code will look simple and we can handle that erroneous scenario in a test case may be . Both are fine.